sdk/core/azure-core/src/main/java/com/azure/core/util/FluxUtil.java

```diff
@@ -270,20 +279,164 @@ public static boolean isFluxByteBuffer(Type entityType) {
      */
     public static <T> Mono<T> withContext(Function<Context, Mono<T>> serviceCall,
         Map<String, String> contextAttributes) {
-        return Mono.deferContextual(context -> {
-            final Context[] azureContext = new Context[]{Context.NONE};
+        return Mono.deferContextual(
+            ctx -> serviceCall.apply(ctx == null || ctx.isEmpty()
+                                         ? Context.NONE : new ReactorToAzureContextWrapper(contextAttributes, ctx)));
+    }
+
+    // This class wraps a reactor context in the Azure-Core Context API, to avoid unnecessary copying
+    private static class ReactorToAzureContextWrapper extends Context {
+        private final Map<String, String> contextAttributes;
+        private final reactor.util.context.ContextView reactorContext;
+        private final int size;
+
+        ReactorToAzureContextWrapper(reactor.util.context.ContextView reactorContext) {
+            this(null, reactorContext);
+        }
+
+        ReactorToAzureContextWrapper(Map<String, String> contextAttributes,
+            reactor.util.context.ContextView reactorContext) {
+            super();
+            this.contextAttributes = contextAttributes;
+            this.reactorContext = reactorContext;
```

Comment:
This will make `com.azure.core.util.Context` mutable because reactor context allows you to delete keys. Previously, Azure Context guaranteed immutability.